### PR TITLE
Fishing Odds Code Improvements and Rescue Hooks

### PR DESCRIPTION
**Pull request by GoldenAlpharex**
## About The Pull Request
I wanted to try and implement an easier way for people to fish out corpses from chasms, as I heard many tales of people trying to fish others out of chasms and it taking over one IRL hour, with some cases where it would take over two hours. Obviously, that's not really interesting gameplay, and it doesn't really give people an incentive to fish, it just turns it into an annoyance that people won't want to do for fun. Now, we don't want that, do we?

As such, I've created the rescue hook, a special fishing hook that can only be used in chasms (as that's currently the only place you can find people into), which will only be able to fish out duds, skeleton corpses, any mob that's fallen into a chasm and hasn't been rescued yet, or rarely, a hostile monster lurking below. It has, at the time of writing this, a weight of 5 (50 without bait, lower with bait) for duds and a weight of 30 for chasm detritus, which themselves have a 50% chance to be a random skeleton corpse, or a lobstrosity, and the remaining 50% chance of fishing out a mob that's fallen into a chasm. I'm open to tweaking these values if we think it's too easy or too hard, but it's still a rather expensive item, so I'd consider it quite fine the way it is myself, as it's still not risk-free.

It's currently only obtainable through buying it from cargo in the goodies section, at a default price of 600 credits (making it SIGNIFICANTLY more expensive than the rest of the fishing content, and making it something that assistants will have to put some elbow grease into if they want to be able to afford it).

As it stands currently, it can't be used to recover the fallen's belongings that weren't on their person (i.e., their crusher if they were holding it in hands), ~*but* I'm down to make that easier to fish out using, for instance, the magnet hook, while also making it incompatible with fishing out bodies, which would make it a nice way to recover those lost items without spending over an hour fishing for them, if that's something that maintainers would want.~ Maintainers did want it, and as such...

The Magnetic hook is now the go-to hook to retrieve objects from chasms! Not only does it inherently do a much better job at fishing out non-fishes, it also has a lesser chance of retrieving random junk from chasms, and an even lower chance of fishing out lobstrosities!

I also improved the code for the fishing weights calculation so that the hooks and the rods can have an effect on the odds of certain types of rewards more easily, with the option of offloading a more of what's currently being calculated on `fishing_challenge` over on the rods or even the hooks themselves.

I finished by fixing a handful of capitalization and punctuation issues in various fishing items, as that bugged me when I was testing my changes.

## Why It's Good For The Game
Corpses being recoverable from chasms was a great idea, however making it so people would have to sink a major portion of their shift for a chance at recovering a corpse doesn't create a particularly interesting gameplay loop. However, being able to spend your hard-earned funds in order to streamline that process without really being able to use that to cheese other mechanics sounds like a great deal to me.

## Changelog

:cl: GoldenAlpharex
add: Added a Rescue Hook, that will allow the fishing rod it's attached onto to become a lot more proficient at recovering corpses from chasms, at the expense of making it unusable for more traditional fishing. It isn't entirely lobstrosity-proof, however...
balance: The magnetic hook can no longer fish out corpses from chasms, but will fish out items much more efficiently than any other hooks, while also being much less attractive to lobstrosities. Some still fall for it regardless, however.
spellcheck: Fixed the capitalization and punctuation in the description of multiple fishing accessories.
code: Improved the code for fishing weights, to allow for different hooks to have some more noticeable results on the weights without having to add to an already massive proc.
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
It's certainly currently much more tedious in practice than I anticipated from testing it locally when developing the initial feature, probably due to the fact that my local tests didn't have quite so many random parts of legions inside the chasms. Definitely makes sense to add a way to make it more reliable.

**Comment by newfren** [comment]
it would be nice if there were some indication that the magnet hook won't work for finding bodies, I can see that leading to frustration

**Review by Twaticus — APPROVED** [review]

File: code/modules/fishing/fishing_equipment.dm [hunk 1]
```diff
@@ -76,6 +122,32 @@
...
+	icon_state = "rescue"
+	fishing_hook_traits = FISHING_HOOK_RESCUE
+	rod_overlay_icon_state = "hook_rescue_overlay"
+	chasm_detritus_type = /obj/item/chasm_detritus/restricted/bodies
+
+
+// This hook can only fish in chasms.
+/obj/item/fishing_hook/rescue/reason_we_cant_fish(datum/fish_source/target_fish_source)
+	if(istype(target_fish_source, /datum/fish_source/chasm))
+		return
```
[the remark this answers is not in the document]
Comment (Mothblocks) [reply code/modules/fishing/fishing_equipment.dm:138]: Or

```suggestion
		return ..()
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/fishing/fish/chasm_detritus.dm [hunk 2]
```diff
@@ -44,3 +68,41 @@
 			chasm_contents += thing
 
 	return chasm_contents
+
+
+/// Variant of the chasm detritus that allows for an easier time at fishing out
+/// bodies, and sometimes less desireable monsters too.
+/obj/item/chasm_detritus/restricted
+	var/chasm_storage_restricted_type = /obj
```
[the remark this answers is not in the document]
Comment (GoldenAlpharex) [reply code/modules/fishing/fish/chasm_detritus.dm:76]: Oh, I did forget, you're right, good catch.

File: code/modules/fishing/fishing_equipment.dm [hunk 3]
```diff
@@ -76,6 +122,32 @@
...
+
+	return "The hook on your fishing rod wasn't meant for traditional fishing, rendering it useless at doing so!"
+
+
+/obj/item/fishing_hook/rescue/get_hook_bonus_multiplicative(fish_type, datum/fish_source/source)
+	// Sorry, you won't catch fish with this.
+	if(ispath(fish_type, /obj/item/fish))
+		return NONE
```
[the remark this answers is not in the document]
Comment (GoldenAlpharex) [reply code/modules/fishing/fishing_equipment.dm:146]: Oh, I didn't know. Fair enough I guess.

**Review by Mothblocks — APPROVED** [review]
@AnturK Give a pass over if you want

**Comment by Elli-Skala** [comment]
> it would be nice if there were some indication that the magnet hook won't work for finding bodies, I can see that leading to frustration

I think this is mostly a issue with fishing as a whole being unclear on what does exactly what

**Comment by GoldenAlpharex** [comment]
> > it would be nice if there were some indication that the magnet hook won't work for finding bodies, I can see that leading to frustration
> 
> I think this is mostly a issue with fishing as a whole being unclear on what does exactly what

Yeah, I agree with this, I might give it another pass to improve its UX later, but I don't have a ton of time to do that at the moment.

**Pull request merged**